instance/models/instance.py

```diff
@@ -526,16 +526,26 @@ def _run_playbook(self, requirements_path, playbook_path):
             playbook_path,
             self.ansible_playbook_filename,
             username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
-        ) as processus:
-            for fd, line in read_files(processus.stdout, processus.stderr):
-                line = line.decode('utf-8').rstrip()
-                if fd == processus.stdout:
-                    self.logger.info(line)
-                elif fd == processus.stderr:
-                    self.logger.error(line)
-                log_lines.append(line)
-            processus.wait()
-            return (log_lines, processus.returncode)
+        ) as process:
+            try:
+                log_line_generator = read_files(
+                    process.stdout,
+                    process.stderr,
+                    line_timeout=settings.ANSIBLE_LINE_TIMEOUT,
+                    global_timeout=settings.ANSIBLE_GLOBAL_TIMEOUT,
```

Comment:
Nice to see that you were able to have both! This will definitely be useful in other cases in the future imho - thanks for looking into it.